pyo3-macros-backend/src/pyfunction.rs

```diff
@@ -469,8 +473,36 @@ fn function_c_wrapper(
                 })
             }
         })
+    } else if spec.can_use_fastcall() {
+        let body = impl_arg_params(spec, None, cb, &py, true)?;
+        Ok(quote! {
+            unsafe extern "C" fn #wrapper_ident(
+                _slf: *mut pyo3::ffi::PyObject,
+                _args: *const *mut pyo3::ffi::PyObject,
+                _nargs: pyo3::ffi::Py_ssize_t,
+                _kwnames: *mut pyo3::ffi::PyObject) -> *mut pyo3::ffi::PyObject
+            {
+                pyo3::callback::handle_panic(|#py| {
+                    #slf_module
+                    // _nargs is the number of positional arguments in the _args array,
+                    // the number of KW args is given by the length of _kwnames
+                    let _kwnames: Option<&pyo3::types::PyTuple> = #py.from_borrowed_ptr_or_opt(_kwnames);
+                    // Safety: &PyAny has the same memory layout as `*mut ffi::PyObject`
+                    let _args = _args as *const &pyo3::PyAny;
+                    let _kwargs = if let Some(kwnames) = _kwnames {
+                        std::slice::from_raw_parts(_args.offset(_nargs), kwnames.len())
+                    } else {
+                        &[]
+                    };
+                    let _args = std::slice::from_raw_parts(_args, _nargs as usize);
```

Comment:
This boilerplate appears twice, perhaps worth refactoring into a `fn fastcall_args_kwargs_boilerplate() -> TokenStream` ?

Reply:
Yeah, I wanted to do this together with adding fastcall support to `static` and `class` methods.  But this is a larger refactoring which I might not have time for soon.

Reply:
No worries; I find some weeks I can do lots of contributions and other weeks I can barely respond to issues. The codebase is gradually marching in the right direction :)